snuba/admin/auth.py

```diff
@@ -0,0 +1,39 @@
+from __future__ import annotations
+
+from typing import Callable
+
+from snuba import state
+
+
+class UnauthorizedException(Exception):
+    pass
+
+
+auth_provider = Callable[[], str]
+
+# This function takes the Flask request and authorizes it.
+# If the request is valid it would return the user id.
+# If not it will raise UnauthorizedException
+#
+# TODO: provide a more structured representation of the User that
+# includes the role at least.
+def authorize_request() -> str:
+    provider_id = state.settings.ADMIN_AUTH_PROVIDER
```

Comment:
Don't think you need to go through state module for this
```suggestion
    provider_id = snuba.settings.ADMIN_AUTH_PROVIDER
```

Reply:
right I changed it from runtime config to settings sand forgot to fix that